Tasks/JavaToolInstallerV0/taskutils.ts

```diff
@@ -0,0 +1,65 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as os from 'os';
+import * as path from 'path';
+
+import { ToolRunner } from 'azure-pipelines-task-lib/toolrunner';
+
+export function sleepFor(sleepDurationInMilliSeconds: number): Promise<any> {
+    return new Promise((resolve, reject) => {
+        setTimeout(resolve, sleepDurationInMilliSeconds);
+    });
+}
+
+/**
+ * Build a path to file in local root.
+ * @param localPathRoot Path to the folder where file should be located.
+ * @param fileNameAndPath Path to the file which name should be taken.
+ * @returns string
+ */
+export function buildFilePath(localPathRoot: string, fileNameAndPath: string): string {
+    const fileName = fileNameAndPath.split(/[\\\/]/).pop();
+    const extractSource = path.join(localPathRoot, fileName);
+
+    return extractSource;
+}
+
+/**
+ * Run a tool with `sudo` on Linux and macOS.
+ * Precondition: `toolName` executable is in PATH.
+ * @returns ToolRunner
+ */
+export function sudo(toolName: string): ToolRunner {
+    if (os.platform() === 'win32') {
+        return tl.tool(toolName);
+    } else {
+        const toolPath = tl.which(toolName);
+        return tl.tool('sudo').line(toolPath);
+    }
+}
+
+/**
+ * Attach a disk image.
+ * Only for macOS.
+ * Returns promise with return code.
+ * @param sourceFile Path to a disk image file.
+ * @returns number
+ */
+export async function attach(sourceFile: string): Promise<number> {
+    console.log(tl.loc('AttachDiskImage'));
+    const hdiutil = sudo('hdiutil');
+    hdiutil.line(`attach "${sourceFile}"`);
+    return await hdiutil.exec();
+}
+
+/**
+ * Detach a disk image.
```

Comment:
Please update it also - since it's also only for 'Only for macOS'

Reply:
Added, thanks!